src/main/java/org/opensearch/flowframework/indices/FlowFrameworkIndicesHandler.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.indices;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.admin.indices.create.CreateIndexRequest;
+import org.opensearch.action.admin.indices.create.CreateIndexResponse;
+import org.opensearch.action.admin.indices.mapping.put.PutMappingRequest;
+import org.opensearch.action.admin.indices.settings.put.UpdateSettingsRequest;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.flowframework.exception.FlowFrameworkException;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.opensearch.flowframework.constant.CommonValue.META;
+import static org.opensearch.flowframework.constant.CommonValue.NO_SCHEMA_VERSION;
+import static org.opensearch.flowframework.constant.CommonValue.SCHEMA_VERSION_FIELD;
+
+/**
+ * Flow Framework Indices Handler
+ */
+public class FlowFrameworkIndicesHandler {
+
+    private static final Logger logger = LogManager.getLogger(FlowFrameworkIndicesHandler.class);
+    private static final Map<String, Object> indexSettings = Map.of("index.auto_expand_replicas", "0-1");
```

Comment:
What's the justification for this `auto_expand_replicas` setting, just trying to understand why this is necessary